zipline/data/_equities.pyx

```diff
@@ -134,7 +135,8 @@ cpdef _read_bcolz_data(ctable_t table,
                        list columns,
                        intp_t[:] first_rows,
                        intp_t[:] last_rows,
-                       intp_t[:] offsets):
+                       intp_t[:] offsets,
+                       read_all):
```

Comment:
Guessing this is untyped because Cython doesn't know about `bool`?  You can type this as `int` and it will behave the way you want, since `bool` is a subclass of `int`.